### PR TITLE
Bug/jenkins 40662 run details close problem

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- Add a regression test for the scenario when a user deep-links to run details with no tab specified in the URL.
- Added a new "waitForLocationChange" custom command that polls for a URL change and only allows the actions to proceed once it changes. Important for things like closing the modal that trigger an async location change
- See [JENKINS-40662](https://issues.jenkins-ci.org/browse/JENKINS-40662).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [x] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

File: src/test/js/edgeCases/runDetailsDeepLink.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                .forRun(1)
+                .waitForElementVisible('@executer');
+        });
+    },
+    /** Check Job Blue Ocean Pipeline Activity Page has run */
+    'Step 03': function (browser) {
+        const blueActivityPage = browser.page.bluePipelineActivity().forJob(jobName, 'jenkins');
+        blueActivityPage.waitForRunRunningVisible(jobName + '-1');
+        const blueRunDetailPage = browser.page.bluePipelineRunDetail().forRun(jobName, 'jenkins', 1);
+        blueRunDetailPage.closeModal('/activity');
```
Comment (scherler) [inline src/test/js/edgeCases/runDetailsDeepLink.js]: @cliffmeyers you need to wait until the run is finished otherwise this can lead to flaky tests :bug:

Comment (cliffmeyers) [inline src/test/js/edgeCases/runDetailsDeepLink.js]: @scherler closing the modal while the job is running can lead to a flaky test? Can you explain why?

File: src/main/js/custom_commands/waitForLocationChange.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/**
+ * @module waitForLocationChange
+ * @memberof custom_commands
```
Comment (cliffmeyers) [inline src/main/js/custom_commands/waitForLocationChange.js]: Fixed in 136d786

File: src/test/js/edgeCases/runDetailsDeepLink.js (hunk 1 above)
Comment (cliffmeyers) [inline src/test/js/edgeCases/runDetailsDeepLink.js]: Fixed in 00843f0

**Pull request merged**